sdk/identity/identity/src/credentials/clientCertificateCredential.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { readFileSync } from "fs";
+import { createHash } from "crypto";
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { IdentityClientOptions, IdentityClient } from "../client/identityClient";
+
+export class ClientCertificateCredential implements TokenCredential {
+  private identityClient: IdentityClient;
+  private _tenantId: string;
+  private _clientId: string;
+  private _certificateString: string;
+
+  public certificateThumbprint: string;
+  public certificateX5t: string;
+
+  constructor(
+    tenantId: string,
+    clientId: string,
+    certificatePath: string,
+    options?: IdentityClientOptions
+  ) {
+    this.identityClient = new IdentityClient(options);
+    this._tenantId = tenantId;
+    this._clientId = clientId;
+
+    this._certificateString = readFileSync(certificatePath, "utf8");
+
+    const certificatePattern = /(-+BEGIN CERTIFICATE-+)(\n\r?|\r\n?)([A-Za-z0-9+/\n\r]+=*)(\n\r?|\r\n?)(-+END CERTIFICATE-+)/;
```

Comment:
> const certificatePattern = /(-+BEGIN CERTIFICATE-+)(\n\r?|\r\n?)([A-Za-z0-9+/\n\r]+=*)(\n\r?|\r\n?)(-+END CERTIFICATE-+)/; [](start = 3, length = 123)

Don't you actually need the private key to sign the tokens sent to AAD?

Reply:
Oh... nm I see you're passing the entire PEM contents as _certificateString

Reply:
Yep, this is just used to generate the public key thumbprint